flutter_vlc_player/ios/Classes/SwiftFlutterVlcPlayerPlugin.swift

```diff
@@ -0,0 +1,1228 @@
+    }
+    
+    public func setStreamUrl(_ input: SetMediaMessage, error: AutoreleasingUnsafeMutablePointer<FlutterError?>) {
+        
+        var player: VLCViewController? = nil
+        if let textureId = input.textureId {
+            player = players[Int(truncating: textureId)]
+        }
+        
+        var isAssetUrl: Bool = false
+        var mediaUrl: String = ""
+        
+        if(DataSourceType(rawValue: Int(truncating: input.type!)) == DataSourceType.ASSET){
+            var assetPath: String
+            if input.packageName != nil {
+                assetPath = registrar.lookupKey(forAsset: input.uri ?? "" , fromPackage: input.packageName ?? "")
+            } else {
+                assetPath = registrar.lookupKey(forAsset: input.uri ?? "")
+            }
+            mediaUrl = assetPath
+            isAssetUrl = true
+        }else{
+            mediaUrl = input.uri ?? ""
+            isAssetUrl = false
+        }
+        player?.setMediaPlayerUrl(
+            uri: mediaUrl,
+            isAssetUrl: isAssetUrl,
+            autoPlay: input.autoPlay?.boolValue ?? true,
+            hwAcc: input.hwAcc?.intValue ?? HWAccellerationType.HW_ACCELERATION_AUTOMATIC.rawValue,
+            options: []
+        )
+    }
+    
+    public func play(_ input: TextureMessage, error: AutoreleasingUnsafeMutablePointer<FlutterError?>) {
+        
+        var player: VLCViewController? = nil
+        if let textureId = input.textureId {
+            player = players[Int(truncating: textureId)]
+        }
+        
+        player?.play()
+    }
+    
+    public func pause(_ input: TextureMessage, error: AutoreleasingUnsafeMutablePointer<FlutterError?>) {
+        
+        var player: VLCViewController? = nil
+        if let textureId = input.textureId {
+            player = players[Int(truncating: textureId)]
+        }
+        
+        player?.pause()
+    }
+    
+    public func stop(_ input: TextureMessage, error: AutoreleasingUnsafeMutablePointer<FlutterError?>) {
+        
+        var player: VLCViewController? = nil
+        if let textureId = input.textureId {
+            player = players[Int(truncating: textureId)]
+        }
+        
+        player?.stop()
+    }
+    
+    public func isPlaying(_ input: TextureMessage, error: AutoreleasingUnsafeMutablePointer<FlutterError?>) -> BooleanMessage? {
+        
+        var player: VLCViewController? = nil
+        if let textureId = input.textureId {
+            player = players[Int(truncating: textureId)]
+        }
+        
+        let message: BooleanMessage = BooleanMessage()
+        message.result = player?.isPlaying()
+        return message
+    }
+    
+    public func setLooping(_ input: LoopingMessage, error: AutoreleasingUnsafeMutablePointer<FlutterError?>) {
+        
+        var player: VLCViewController? = nil
+        if let textureId = input.textureId {
+            player = players[Int(truncating: textureId)]
+        }
```

Comment:
This piece repeats a lot in methods here. Perhaps worth extracting it into a getter?

Reply:
I tried this tonight, the issue is that the "input" variable has different types based on each method. Ie "_ input: TextureMessage",  "input: PositionMessage", "input: LoopingMessage" 

Im all for D.R.Y but not entirely sure how to manage here without using some sort of overkill generics. Thoughts?

Reply:
We're grabbing the input by the textureId field everywhere, right? 
```swift
func getPlayer(textureId: Float?) -> VLCViewController? {
  guard textureId != nil else { return nil }
  return players[Int(truncating: textureId)]
}
```

Reply:
Swift is not my native language :) I pushed up a change to review.